pkg/kubectl/cmd/create_job.go

```diff
@@ -74,74 +70,68 @@ func NewCmdCreateJob(f cmdutil.Factory, cmdOut io.Writer) *cobra.Command {
 	cmdutil.AddApplyAnnotationFlags(cmd)
 	cmdutil.AddValidateFlags(cmd)
 	cmdutil.AddPrinterFlags(cmd)
-	cmd.Flags().String("from-cronjob", "", "Specify the name of the CronJob to create a Job from.")
+	cmd.Flags().String("from", "", "The name of the resource to create a Job from (only cronjob is supported).")
 
 	return cmd
 }
 
 func (c *CreateJobOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) (err error) {
-	c.FromCronJob = cmdutil.GetFlagString(cmd, "from-cronjob")
-
-	// Complete other options for Run.
-	c.Mapper, _ = f.Object()
-
-	c.OutputFormat = cmdutil.GetFlagString(cmd, "output")
+	if len(args) == 0 {
+		return cmdutil.UsageErrorf(cmd, "NAME is required")
+	}
+	c.Name = args[0]
 
+	from := cmdutil.GetFlagString(cmd, "from")
 	c.Namespace, _, err = f.DefaultNamespace()
 	if err != nil {
 		return err
 	}
+	builder := f.NewBuilder().
+		Unstructured().
+		NamespaceParam(c.Namespace).DefaultNamespace().
+		ResourceTypeOrNameArgs(false, from).
+		Flatten().
+		Latest()
+	infos, err := builder.Do().Infos()
```

Comment:
calls to the server are discouraged/disallowed in complete because of fears over local